pkg/providers/snow/reconciler/reconciler.go

```diff
@@ -2,19 +2,151 @@ package reconciler
 
 import (
 	"context"
+	"fmt"
+	"os"
+	"time"
 
 	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	c "github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/constants"
 	"github.com/aws/eks-anywhere/pkg/controller"
+	"github.com/aws/eks-anywhere/pkg/controller/clientutil"
+	"github.com/aws/eks-anywhere/pkg/controller/serverside"
+	"github.com/aws/eks-anywhere/pkg/providers/snow"
 )
 
-type Reconciler struct{}
+const defaultRequeueTime = time.Minute
 
-func New() *Reconciler {
-	return &Reconciler{}
+const (
+	controlSpecPlaneAppliedCondition clusterv1.ConditionType = "ControlPlaneSpecApplied"
+	controlPlaneReadyCondition       clusterv1.ConditionType = "ControlPlaneReady"
+	eksaLicense                      string                  = "EKSA_LICENSE"
+)
+
+type Reconciler struct {
+	client     client.Client
+	validator  *snow.AwsClientValidator
+	defaulters *snow.Defaulters
+}
+
+func New(client client.Client, validator *snow.AwsClientValidator, defaulters *snow.Defaulters) *Reconciler {
+	return &Reconciler{
+		client:     client,
+		validator:  validator,
+		defaulters: defaulters,
+	}
 }
 
-func (v *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (controller.Result, error) {
+func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (controller.Result, error) {
+	dataCenterConfig := &anywherev1.SnowDatacenterConfig{}
+	dataCenterName := types.NamespacedName{Namespace: cluster.Namespace, Name: cluster.Spec.DatacenterRef.Name}
+	if err := r.client.Get(ctx, dataCenterName, dataCenterConfig); err != nil {
+		return controller.Result{}, err
+	}
+
+	if err := setupEnvVars(ctx, r.client); err != nil {
+		log.Error(err, "Failed to set up snow env vars")
+		return controller.Result{}, err
+	}
+
+	if !dataCenterConfig.Status.SpecValid {
```

Comment:
So based on @g-gaston comment above, we might not even need this check? But we should check for the machineconfig too